persistence/src/main/java/org/mskcc/cmo/metadb/persistence/neo4j/MetadbSampleRepository.java

```diff
@@ -20,11 +20,21 @@
     @Query("MATCH (sm: Sample {metaDbSampleId: $metaDbSampleId}) "
             + "RETURN sm")
     MetadbSample findAllSamplesById(@Param("metaDbSampleId") UUID metaDbSampleId);
-    
+
+    @Query("MATCH (s:Sample)-[:HAS_METADATA]->(s: SampleMetadata {primaryId: $primaryId})"
+            + "RETURN s")
+    MetadbSample findSampleByPrimaryId(@Param("primaryId") String primaryId);
+
     @Query("MATCH (s: SampleAlias {value: $igoId.sampleId, namespace: 'igoId'}) "
         + "MATCH (s)<-[:IS_ALIAS]-(sm: Sample) "
         + "RETURN sm")
-    MetadbSample findResearchSampleByIgoId(@Param("igoId") SampleAlias igoId); 
+    MetadbSample findResearchSampleByIgoId(@Param("igoId") SampleAlias igoId);
+
+    //needs to be tested with clinical samples
+    @Query("MATCH (s: SampleAlias {value: $dmpId.sampleId, namespace: 'dmpId'}) "
+            + "MATCH (s)<-[:IS_ALIAS]-(sm: Sample) "
+            + "RETURN sm")
+    MetadbSample findClinicalSampleByDmpId(@Param("dmpId") SampleAlias dmpId);
```

Comment:
Same comment here I think where we can consolidate these two methods with a single method

```
@Query("MATCH (s: SampleAlias {value: $sampleAlias.value, namespace: $sampleAlias.namespace}) "
        + "MATCH (s)<-[:IS_ALIAS]-(sm: Sample) "
        + "RETURN sm")
MetadbSample findMetadbSampleByAlias(@Param("sampleAlias") SampleAlias sampleAlias);
```